src/components/Context/index.js

```diff
@@ -1,79 +1,96 @@
 import React, {Component} from 'react';
 
+const uuidv4 = require('uuid/v4');
 const CarrotBagContext = React.createContext();
 
 export class Provider extends Component {
-    state = {
+  constructor(props) {
```

Comment:
Random question- Why have you used a constructor here? It works without the constructor so not sure why we need it, unless its best practice, but seems like extra code we don't need.